Python/Include/Range.hpp

```diff
@@ -0,0 +1,81 @@
+#ifndef ROVER_PYTHON_RANGE_HPP
+#define ROVER_PYTHON_RANGE_HPP
+#include <string_view>
+#include <type_traits>
+#include <variant>
+#include <pybind11/pybind11.h>
+#include "Arithmetics.hpp"
+#include "Rover/Box.hpp"
+#include "Rover/Range.hpp"
+
+namespace Rover {
+  namespace Details {
+    class PythonRange {
+      public:
+        using Type = Range<pybind11::object, pybind11::object>::Type;
+        using ContinuousRange = Range<pybind11::object, pybind11::object>;
+        using DiscreteRange = Range<pybind11::object, pybind11::object,
+          pybind11::object>;
+
+        template<typename ImplFwd, std::enable_if_t<std::is_convertible_v<
+          std::decay_t<ImplFwd>, ContinuousRange>>* = nullptr>
+        PythonRange(ImplFwd&& impl)
+          : m_impl(std::forward<ImplFwd>(impl)) {}
+
+        template<typename ImplFwd, std::enable_if_t<std::is_convertible_v<
+          std::decay_t<ImplFwd>, DiscreteRange>>* = nullptr>
+        PythonRange(ImplFwd&& impl)
+          : m_impl(std::forward<ImplFwd>(impl)) {}
+
+        PythonRange(pybind11::object begin, pybind11::object end)
+          : m_impl(Range(std::move(begin), std::move(end))) {}
+
+        PythonRange(pybind11::object begin, pybind11::object end,
+            pybind11::object granularity)
+          : m_impl(Range(std::move(begin), std::move(end),
```

Comment:
You'll need to box ```begin```, ```end```, and ```granularity```.

Reply:
Boxing is implicit and it is done on the C++ side, isn't it?

Reply:
Oh ok, it won't work if I pass sub-generators rather than values. Let me fix it.